pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -544,13 +544,15 @@ func (s *DiscoveryServer) pushConnection(con *XdsConnection, pushEv *XdsEvent) e
 		return nil
 	}
 
-	if err := con.modelNode.SetWorkloadLabels(s.Env); err != nil {
+	if err := con.modelNode.SetServiceInstances(pushEv.push.Env); err != nil {
```

Comment:
Can we have a test for this change as well? Preferably, it would be good to have a test which would fail before this change, but passes after ... if possible.

Reply:
I thought about it , it is hard to simulate a so deeper calling stack.

Reply:
Surly we can find a way to test it with a unit test? If we're relying on order here, we have to make sure that the order is enforced ... otherwise someone might break it in the future.